src/PhpPact/Standalone/ProviderVerifier/Verifier.php

```diff
@@ -2,264 +2,198 @@
-            }
-        }
-
-        if (\count($this->config->getProviderVersionTag()) > 0) {
-            foreach ($this->config->getProviderVersionTag() as $tag) {
-                $parameters[] = "--provider-version-tag={$tag}";
-            }
-        }
-
-        if ($this->config->getProviderStatesSetupUrl() !== null) {
-            $parameters[] = "--provider-states-setup-url={$this->config->getProviderStatesSetupUrl()}";
-        }
-
-        if ($this->config->isPublishResults() === true) {
-            $parameters[] = '--publish-verification-results';
-        }
-
-        if ($this->config->getBrokerToken() !== null) {
-            $parameters[] = "--broker-token={$this->config->getBrokerToken()}";
-        }
-
-        if ($this->config->getBrokerUsername() !== null) {
-            $parameters[] = "--broker-username={$this->config->getBrokerUsername()}";
-        }
-
-        if ($this->config->getBrokerPassword() !== null) {
-            $parameters[] = "--broker-password={$this->config->getBrokerPassword()}";
-        }
-
-        if (count($this->config->getCustomProviderHeaders()) > 0) {
-            foreach ($this->config->getCustomProviderHeaders() as $customProviderHeader) {
-                $parameters[] = "--custom-provider-header=\"{$customProviderHeader}\"";
-            }
-        }
+        $this->handle = $this->client->call(
+            'pactffi_verifier_new_for_application',
+            $config->getCallingApp()->getName(),
+            $config->getCallingApp()->getVersion()
+        );
+    }
 
-        if ($this->config->isVerbose() === true) {
-            $parameters[] = '--verbose=VERBOSE';
-        }
+    private function setProviderInfo(VerifierConfigInterface $config): void
+    {
+        $this->client->call(
+            'pactffi_verifier_set_provider_info',
+            $this->handle,
+            $config->getProviderInfo()->getName(),
+            $config->getProviderInfo()->getScheme(),
+            $config->getProviderInfo()->getHost(),
+            $config->getProviderInfo()->getPort(),
+            $config->getProviderInfo()->getPath()
+        );
+    }
 
-        if ($this->config->getLogDirectory() !== null) {
-            $parameters[] = "--log-dir={$this->config->getLogDirectory()}";
+    private function setProviderTransports(VerifierConfigInterface $config): void
+    {
+        foreach ($config->getProviderTransports() as $transport) {
+            $this->client->call(
+                'pactffi_verifier_add_provider_transport',
+                $this->handle,
+                $transport->getProtocol(),
+                $transport->getPort(),
+                $transport->getPath(),
+                $transport->getScheme()
+            );
         }
+    }
 
-        if ($this->config->getFormat() !== null) {
-            $parameters[] = "--format={$this->config->getFormat()}";
-        }
+    private function setFilterInfo(VerifierConfigInterface $config): void
```

Comment:
`setProviderState` and `setFilterInfo` function names got interchanged I think. Might not affect functionality but since I was reading this implementation thought should point this to you :)

Reply:
You are correct. Nice catch. I will create a PR to update it.

Reply:
Fixed by https://github.com/pact-foundation/pact-php/pull/542